serializers/src/glTF/2.0/babylon.glTFMaterialExporter.ts

```diff
@@ -500,20 +477,56 @@ module BABYLON.GLTF2 {
          * @param mimeType mimetype of the image
          * @returns base64 image string
          */
-        private static _CreateBase64FromCanvas(buffer: Uint8ClampedArray | Float32Array, width: number, height: number, mimeType: ImageMimeType): string {
-            const imageCanvas = document.createElement('canvas');
-            imageCanvas.width = width;
-            imageCanvas.height = height;
-            imageCanvas.id = "WriteCanvas";
-
-            const ctx = imageCanvas.getContext('2d') as CanvasRenderingContext2D;
-
-            const imgData = ctx.createImageData(width, height);
+        private _createBase64FromCanvasAsync(buffer: Uint8Array | Float32Array, width: number, height: number, mimeType: ImageMimeType): Promise<string> {
+            this._createBase64FromCanvasPromiseChain = this._createBase64FromCanvasPromiseChain.then(() => {

```

Comment:
do you still need to chain the promises?

Reply:
Nope, I removed the chained promises.